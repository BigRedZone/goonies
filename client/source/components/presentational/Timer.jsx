import React from 'react';

class Timer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      elapsedTime: 0,
      stop: null,
      time: null,
    };
    this.startTimer = this.startTimer.bind(this);
    this.stopTimer = this.stopTimer.bind(this);
    this.resetTimer = this.resetTimer.bind(this);
  }

  startTimer() {
    this.time = setInterval(() => this.setState({
      elapsedTime: this.state.elapsedTime + 1,
    }),
    1000);
  }

  stopTimer() {
    clearInterval(this.time);
    this.setState({
      stop: this.time
    });
  }

  resetTimer() {
    const { time } = this.state;
    clearInterval(time);
    this.setState({
      elapsedTime: 0,
    });
  }

  render() {
    const timeFormat = sec => (`${Math.floor(sec / 60)}:${(`0${sec % 60}`).slice(-2)}`);

    const Button = props => (<button type="button" {...props} />);
    const { elapsedTime } = this.state;
    return (
      <div>
        <div>
          <h1>
            {timeFormat(elapsedTime)}
          </h1>
          <Button onClick={this.startTimer}>Start</Button>
          <Button onClick={this.stopTimer}>Stop</Button>
          <Button onClick={this.resetTimer}>Reset</Button>
        </div>
      </div>
    );
  }
}

export default Timer;
