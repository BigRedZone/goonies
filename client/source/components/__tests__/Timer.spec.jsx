import React from 'react';
import renderer from 'react-test-renderer';
import Timer from '../presentational/Timer';

test('App Renders correctly', () => {
  const component = renderer.create(<Timer />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
