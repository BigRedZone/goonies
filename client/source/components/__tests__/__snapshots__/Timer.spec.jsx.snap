// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Renders correctly 1`] = `
<div>
  <div
    className="timer-center"
  >
    <h1
      className="timer-format"
    >
      0:00
    </h1>
    <button
      className="timer-btns start-btn"
      onClick={[Function]}
      type="button"
    >
      Start
    </button>
    <button
      className="timer-btns stop-btn"
      onClick={[Function]}
      type="button"
    >
      Stop
    </button>
    <button
      className="timer-btns start-btn"
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
    <button
      className="timer-btns stop-btn"
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
    <br />
    <div
      className="fieldwrapper"
    >
      <div
        className="timer-text"
      >
         Distance Hiked (miles): 
      </div>
      <input
        className="distance-calculator"
        name="distanceInMiles"
        onChange={[Function]}
        placeholder="miles"
        type="text"
        value={0}
      />
      <div
        className="timer-text"
      >
         Total (miles): 
      </div>
      <input
        className="distance-calculator"
        name="trailDistance"
        onChange={[Function]}
        placeholder="miles"
        type="text"
        value={0}
      />
    </div>
  </div>
  <ul
    className="timer-save"
  />
</div>
`;
